src/session_impl.cpp

```diff
@@ -1320,7 +1320,7 @@ namespace aux {
 #ifdef TORRENT_USE_OPENSSL
 		// SSL torrents use the SSL listen port
 		if (req.ssl_ctx) req.listen_port = ssl_listen_port();
-		req.ssl_ctx = &m_ssl_ctx;
+		else req.ssl_ctx = &m_ssl_ctx;
```

Comment:
the previous logic here seemed suspicious too. surely if the torrent has a cert, that's what we should authenticate the tracker against